Applications/shapeworks/MeshCommands.cpp

```diff
@@ -1033,62 +1031,62 @@ void WarpMesh::buildParser()
-
+  parser.add_option("--target_points").action("store").type("multistring").set_default("").help("Names of target points (must be followed by `--`), ex: \"... --target_points *.particles -- ...");
+  parser.add_option("--save_dir").action("store").type("string").set_default("").help("Optional: Path to the directory where the mesh files will be saved");
   Command::buildParser();
 }
-
 bool WarpMesh::execute(const optparse::Values &options, SharedCommandData &sharedData)
 {
   std::string inputMeshFilename = options["reference_mesh"];
   if (inputMeshFilename.length() == 0) {
     std::cerr << "warpmesh error: no reference mesh specified, must pass `--reference_mesh <filename>`\n";
     return false;
   }
-
   std::string inputPointsFilename = options["reference_points"];
   if (inputPointsFilename.length() == 0) {
     std::cerr << "warpmesh error: no reference points specified, must pass `--reference_points <filename>`\n";
     return false;
   }
-
-  std::string targetMeshFilename = options["target_mesh"];
-  if (targetMeshFilename.length() == 0) {
-    std::cerr << "warpmesh error: no target mesh specified, must pass `--target_mesh <filename>`\n";
-    return false;
-  }
-
-  std::string targetPointsFilename = options["target_points"];
-  if (targetPointsFilename.length() == 0) {
-    std::cerr << "warpmesh error: no target points specified, must pass `--target_points <filename>`\n";
+  std::vector<std::string> targetPointsFilenames = options.get("target_points");
+  if (targetPointsFilenames.size() == 0) {
+    std::cerr << "warpmesh error: no target points specified, must pass `--target_points <filenames> --`\n";
     return false;
   }
 
+  std::string saveDir = options["save_dir"];
   try {
     Mesh inputMesh(inputMeshFilename);
-
-    std::vector<std::string> paths;
-    paths.push_back(inputPointsFilename);
-    paths.push_back(targetPointsFilename);
-    ParticleSystem particlesystem(paths);
+    targetPointsFilenames.push_back(inputPointsFilename);
+    ParticleSystem particlesystem(targetPointsFilenames);
     Eigen::MatrixXd allPts = particlesystem.Particles();
-    Eigen::MatrixXd staticPoints = allPts.col(0);
-    Eigen::MatrixXd movingPoints = allPts.col(1);
+    Eigen::MatrixXd staticPoints = allPts.col(targetPointsFilenames.size() - 1);
     int numParticles = staticPoints.rows() / 3;
     staticPoints.resize(3, numParticles);
     staticPoints.transposeInPlace();
-    movingPoints.resize(3, numParticles);
-    movingPoints.transposeInPlace();
-
     MeshWarper warper;
     warper.set_reference_mesh(inputMesh.getVTKMesh(), staticPoints);
-    Mesh output = warper.build_mesh(movingPoints);
-    output.write(targetMeshFilename);
-
+    std::string infnm;
+    
+    if (saveDir.length() > 0) {
+      boost::filesystem::create_directories(saveDir);
+    }
+
+    for (int i = 0; i < targetPointsFilenames.size() - 1; i++) {
+      infnm = targetPointsFilenames[i];
+      infnm.replace(infnm.end()-9, infnm.end(), "vtk");
```

Comment:
I'm nervous about assuming input files end with `particles`. Maybe check this or simply replace everything after the last `.` with `vtk` instead of `-9`.

Reply:
ok makes sense I'll make that change.